code/obj/item/critter_holder.dm

```diff
@@ -0,0 +1,150 @@
+				user.visible_message("<span class='notice'>Just as [user] puts \the [src.held_mobcritter] into \the [src.loc], \the [src.held_mobcritter] jumps right back out!</span>",\
+				"<span class='notice'>[prob(1) && prob(1) ? "You try to put [src.held_mobcritter] into [src.loc], but it hasn't been implemented yet! It jumps free of your grip." : "The moment you let go of \the [src.held_mobcritter], it jumps away!</span>"]")
+			src.unshellify()
+		else
+			src.update_icon()
+
+	/// Handles just about everything related to stuffing a critter into a shell and placing it in someone's hand
+	proc/shellify_critter(mob/living/critter/mob_critter, mob/living/grabber)
+		if(!istype(mob_critter) || !grabber)
+			src.unshellify()
+			return FALSE
+
+		src.icon = mob_critter.icon
+		src.icon_state = mob_critter.icon_state
+		src.name = mob_critter.name
+		src.desc = mob_critter.desc
+		mob_critter.grabber = grabber
+		mob_critter.metaholder = src
+		src.held_mobcritter = mob_critter
+		src.inhand_image_icon = mob_critter.icon_inhand
+		src.item_state = mob_critter.item_state_inhand
+		src.update_icon()
+
+		if(!grabber.put_in_hand(src)) // If they can't hold this, then they can't hold the critter
+			grabber.u_equip(src)
+			boutput(grabber, "<span class='alert'>You can't seem to get a good grip on [mob_critter], try emptying your hands.</span>")
+			src.unshellify()
+			return FALSE
+
+		mob_critter.set_loc(src)
+		RegisterSignal(src.held_mobcritter, list(COMSIG_ATOM_DIR_CHANGED, COMSIG_MOB_DEATH), .proc/update_icon)
+		src.showoff_word = pick(src.showoff_adj)
+		return TRUE // it worked!
+
+	/// Handles safe release of the critter and their leavings, then self-destructs. Also returns the mob, for throwing purposed
+	proc/unshellify()
+		if(length(src.contents))
+			for(var/atom/movable/AM in src.contents)
+				AM.set_loc(get_turf(src))
+		if(src.held_mobcritter)
+			. = src.held_mobcritter
+			UnregisterSignal(src.held_mobcritter, list(COMSIG_ATOM_DIR_CHANGED, COMSIG_MOB_DEATH, COMSIG_MOVABLE_MOVED))
+			src.held_mobcritter.metaholder = null
+			src.held_mobcritter.grabber = null
+			src.held_mobcritter.set_loc(get_turf(src))
+			src.held_mobcritter = null
+		qdel(src)
+
+	/// Makes the object look more like the critter when it moves
+	proc/update_icon()
+		if(src.held_mobcritter)
+			src.name = src.held_mobcritter.name
+			src.desc = src.held_mobcritter.desc
+			src.icon = src.held_mobcritter.icon
+			src.icon_state = src.held_mobcritter.icon_state
+			var/image/critter_overlays = image(src.held_mobcritter.icon, src.held_mobcritter.icon_state)
+			var/image/overlay_eyes = src.held_mobcritter.GetOverlayImage("eyes")
+			if(overlay_eyes)
+				critter_overlays.overlays += overlay_eyes
+			var/image/overlay_hair = src.held_mobcritter.GetOverlayImage("hair")
+			if(overlay_hair)
+				critter_overlays.overlays += overlay_hair
+			src.UpdateOverlays(critter_overlays, "critter_overlays")
```

Comment:
why not .appearance?

Reply:
I tried that, but the item didn't show up in hand. Weird thing was, it showed up if right clicked, and the item mouseover ghost thing showed it. I thought it might've been a layering issue, so i made it set the layer to the hud layer, then to 999, both with no change. I also tried setting the item to have the mob as it's vis_contents, and the item was invisible again until I moved it to my pocket, and it wouldn't update the appearance when the mob died.